Sources/TABResourceLoader/Services/NetworkResponseHandler.swift

```diff
@@ -28,6 +28,11 @@ struct NetworkResponseHandler {
       return .failure(NetworkServiceError.noData)
     }
 
-    return resource.result(from: data)
+    do {
+      let model = try resource.model(from: data)
+      return .success(model)
+    } catch {
+      return .failure(NetworkServiceError.networkingError(error: error))
```

Comment:
is this right to be throwing network service error here even though the network request was probably ok, how come we're not just returning `.failure(error)`?